### PR TITLE
Refactor should eager load

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1035 
| License       | MIT
| Doc PR        | na

### CODE REVIEW

**Comment by soyuka** [comment]
please don't bother reviewing this until I put RFR still thinking :).

**Comment by soyuka** [comment]
Coverage thing is kinda weird, the 3 modified files have 100% coverage -_-.

screen shot 2017-04-29 at 22 46 19

File: src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php [hunk 1]
```diff
@@ -35,6 +36,8 @@
  */
 final class EagerLoadingExtension implements QueryCollectionExtensionInterface, QueryItemExtensionInterface
 {
+    use ShouldEagerLoad;
```
Comment (teohhanhui) [inline src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php]: I thought we don't like traits anymore lol... And the naming is just weird. :stuck_out_tongue:

Comment (soyuka) [inline src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php]: :stuck_out_tongue: I never liked traits, if you have another solution/name in mind please tell me I'll do another pull request ;)

Comment (teohhanhui) [inline src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php]: On another note, using a `Trait` suffix is still the Symfony CS, right?

Maybe `EagerLoadingTrait` \o/

**Pull request merged**